tools/check-messages.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © 2023 Pierre Le Marre <dev@wismill.eu>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the next
+ * paragraph) shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include "config.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "messages.h"
+
+static void
+usage(char **argv)
+{
+    printf("Usage: %s MESSAGE_CODES\n"
+           "\n"
+           "Check whether the given message codes are supported.\n",
+           argv[0]);
+}
+
+#define XKB_CHECK_MSG_ERROR_PREFIX "xkb-check-messages: ERROR: "
+
+static xkb_message_code_t
+parse_message_code(char *raw_code) {
+    xkb_message_code_t code;
+    code = atoi(raw_code);
+    if (!code && strstr(raw_code, "XKB-")) {
+        return atoi(&(raw_code[4]));
+    } else {
+        return code;
+    }
+}
+
+int main(int argc, char **argv) {
+    if (argc <= 1) {
+        usage(argv);
+        return EXIT_INVALID_USAGE;
+    }
+
+    int rc = 0;
+    xkb_message_code_t code;
+    const struct xkb_message_entry* entry;
+    for (int k=1; k < argc; k++) {
+        code = parse_message_code(argv[k]);
+        if (!code) {
+            fprintf(stderr,
+                    XKB_CHECK_MSG_ERROR_PREFIX "Malformed message code: %s\n",
+                    argv[k]);
+            rc |= 0b01;
+            continue;
+        }
+        entry = xkb_message_get(code);
+        if (entry == NULL) {
+            fprintf(stderr,
+                    XKB_CHECK_MSG_ERROR_PREFIX "Unsupported message code: %s\n",
+                    argv[k]);
+            rc |= 0b10;
```

Comment:
you and your new-fangled features :)

tbh, in this case I don't think it's better than just a `rc |= 2;`, binary notation for me doesn't really become useful until we hit 4 binary digits or more. Not going to fight this too hard though (though I expect there'll be *some* system somewhere that will complain)

Reply:
Done